### ISSUE TITLE
WP webservice does not return KNApSAcK Xrefs for pathways

### ISSUE THREAD

**Issue by egonw** [wikipathways/wikipathways.org#20]
As reported [here](https://www.wikipathways.org/index.php/Pathway_Talk:WP3620#lqt_thread_659) and with a test:

```
curl -X GET --header "Accept: application/json" "https://webservice.wikipathways.org/getXrefList?pwId=WP3620&code=Cks&format=json"
```

This is what the Webservice UI gives:

![image](https://user-images.githubusercontent.com/26721/33308098-743db76a-d419-11e7-82d7-c05ce4899fb3.png)

@JonathanMELIUS, do you have ideas about this? Is the webservice using the latest BridgeDb 2.2 and metabolite mapping data?

**Comment by egonw** [comment]
Depends on https://github.com/wikipathways/lucene-indexer/issues/2

**Comment by egonw** [comment]
@mkutmon, is there something ready to test now?

**Comment by egonw** [comment]
ping @JonathanMELIUS @mkutmon, what is the state of this issue? If I run the curl command, it seems it is still not returning someting, but the linked lucene-indexer issue is marked closed?

**Comment by JonathanMELIUS** [comment]
@egonw 
I don't know how the WikiPathways webservice works in details. But the issue seems to definitely be the BridgeDb version used there, an old datasource.txt file. 

The pathways is indexed and the webservice uses the metabolite database since we get some results to those calls:
https://webservice.wikipathways.org/getXrefList?pwId=WP3620&code=Ca
https://webservice.wikipathways.org/getXrefList?pwId=WP3620&code=Ch

**Comment by AlexanderPico** [comment]
@ariutta Can you identify the source of BridgeDb information for the latest Annotation Panel (and datanode table linkouts) on pathway pages?

**Comment by AlexanderPico** [comment]
We should make a plan to consolidate the sources of BridgeDb info (including the datasources.txt). There were as many as 3 at one point. Not sure how many there are now. Ideally, there would be one.

**Comment by nunogit** [comment]
I followed all the webservice stack and didn't find anything that could create issues with the call.
I think the problem might be that the service at http://vm4.wikipathways.org:8983/ is outdated. That's the server that needs to be updated.  I briefly checked this in the afternoon with Jonathan but I didn't realized that this was the server that was being called.

**Comment by ariutta** [comment]
> Can you identify the source of BridgeDb information for the latest Annotation Panel (and datanode table linkouts) on pathway pages?

@AlexanderPico right now, the Annotation Panel and datanode table linkouts are both using the [XrefPanel.php](https://github.com/wikipathways/wikipathways.org/blob/ab8235729355d2d331a1cff8518cebea1849c798/wpi/extensions/XrefPanel.php) extension.

How about we always make the latest datasources.txt and datasources_headers.txt available from bridgedb.org? We'll need to be sure bridgedb.org has [CORS enabled](https://enable-cors.org/) for those files, so that JS resources at wikipathways.org can access them.

**Comment by egonw** [comment]
@AlexanderPico, yes, we should aim at consolidating... for now, updating what we have sounds useful to me too.

@ariutta, I'll file a feature request for the BridgeDb webservice to provide more info about data sources...

**Comment by JonathanMELIUS** [comment]
@egonw @AlexanderPico @ariutta,

@mkutmon show us that it is not a code issue but an outdated metabolite database issue:
https://webservice.wikipathways.org/getXrefList?pwId=WP7&code=Cks

The webservice returns KNApSAcK (Cks) ids for the WP7.

And on https://www.wikipathways.org/index.php/Pathway:WP3620, the Cks ids do map in the "External references" table

So basically, we just need to update the databases on the index server (vm4.wikipathways.org).

**Comment by JonathanMELIUS** [comment]
I think to be more consistent all the ids mapping in WikiPathways servers ( index, webservice, pvjs, external references table, ... ) should use the same instance of BridgeDb in vm4.wikipathways.org

Then webservice.bridgedb.org would be free from WikiPathways servers traffic ??

**Comment by kozo2** [comment]
I am trying to use Wikipathways as a data source of our (@afukushima and me) Metabolite Set Enrichment Analysis software.
I need your cooperation for that.
I'm sorry to trouble you, but I'd like you to try updating vm4.
Would it be difficult?

**Comment by AlexanderPico** [comment]
I thought it was updated...  @nunogit @JonathanMELIUS did your account access work?

**Comment by kozo2** [comment]
@AlexanderPico It seems that the update has no effect about this issue yet.
https://webservice.wikipathways.org/getXrefList?pwId=WP3620&code=Cks

**Comment by AlexanderPico** [comment]
@egonw Do you know where the fix to this issue lies? Should the updated databases address this?

**Comment by nunogit** [comment]
@AlexanderPico no, I was not able to access the server. I didn't get any feedback from Jonathan,
